lib/src/rules/missing_whitespace_between_adjacent_strings.dart

```diff
@@ -90,27 +90,39 @@ class _Visitor extends RecursiveAstVisitor<void> {
     if (string is SimpleStringLiteral) {
       return test([string.value]);
     } else if (string is StringInterpolation) {
-      return test(string.elements.map((e) {
-        if (e is InterpolationString) return e.value;
-        return '';
-      }));
+      var interpolationSubstitutions = <String>[];
+      for (var e in string.elements) {
+        // Given a [StringInterpolation] like '$text', the elements include
+        // empty [InterpolationString]s on either side of the
+        // [InterpolationExpression]. Don't include them in the evaluation.
+        if (e is InterpolationString && e.value.isNotEmpty) {
+          interpolationSubstitutions.add(e.value);
+        }
+        if (e is InterpolationExpression) {
+          // Treat an interpolation expression as a string with whitespace. This
+          // prevents over-reporting on adjascent Strings which start or end
+          // with interpolations.
+          interpolationSubstitutions.add(' ');
+        }
+      }
+      return test(interpolationSubstitutions);
     }
     throw ArgumentError('${string.runtimeType}: $string');
   }
 
-  bool _hasWhitespace(String value) => _whitespaces.any(value.contains);
-  bool _endsWithWhitespace(String value) => _whitespaces.any(value.endsWith);
-  bool _startsWithWhitespace(String value) =>
-      _whitespaces.any(value.startsWith);
-
   static bool _isRegExpInstanceCreation(AstNode? node) {
     if (node is InstanceCreationExpression) {
       var constructorElement = node.constructorName.staticElement;
-      return constructorElement != null &&
-          constructorElement.enclosingElement.name == 'RegExp';
+      return constructorElement?.enclosingElement.name == 'RegExp';
     }
     return false;
   }
 }
 
+extension on String {
+  bool get hasWhitespace => _whitespaces.any(contains);
+  bool get endsWithWhitespace => _whitespaces.any(endsWith);
+  bool get startsWithWhitespace => _whitespaces.any(startsWith);
+}
+
 const _whitespaces = [' ', '\n', '\r', '\t'];
```

Comment:
If this is now only used in the extension, consider making it a static member of the extension (and dropping the leading underscore).

Reply:
done